craft_parts/plugins/ruby_plugin.py

```diff
@@ -0,0 +1,207 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright 2025 Canonical Ltd.
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License version 3 as published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The Ruby plugin."""
+
+import logging
+from enum import Enum
+from typing import Literal
+
+from typing_extensions import override
+
+from .base import Plugin
+from .properties import PluginProperties
+
+logger = logging.getLogger(__name__)
+
+
+class RubyFlavor(str, Enum):
+    """All Ruby implementations supported by ruby-install."""
+
+    ruby = "ruby"
+    jruby = "jruby"
+    rbx = "rbx"
+    truffleruby = "truffleruby"
+    mruby = "mruby"
+
+
+class RubyPluginProperties(PluginProperties, frozen=True):
+    """The part properties used by the Ruby plugin."""
+
+    plugin: Literal["ruby"] = "ruby"
+    source: str  # pyright: ignore[reportGeneralTypeIssues]
+
+    ruby_gems: list[str] = []
+    ruby_use_bundler: bool = False
+
+    # build arguments
+    ruby_flavor: RubyFlavor = RubyFlavor.ruby
+    ruby_version: str = "3.2"
+    ruby_prefix: str = "/usr"
+    ruby_use_jemalloc: bool = False
+    ruby_shared: bool = False
+    ruby_configure_options: list[str] = []
+
+
+class RubyPlugin(Plugin):
+    """A plugin for Ruby based projects.
+
+    The desired Ruby interpreter is compiled using ruby-install.
+
+    The ruby plugin uses the common plugin keywords, plus the following ruby-
```

Comment:
> common plugin keywords

A minor point, but there aren't really any common keys for plugins. Plugin keys are parallel with part keys, so they feel like they're part of the same collection, but there's no standard set, AFAIK.